### ISSUE TITLE
Stack overflow from cljd.compiler$load_libs_info

### ISSUE THREAD

**Issue by perpen** [Tensegritics/ClojureDart#126]
I just followed both quickstarts - to the letter afaik. The dart one worked fine, but the flutter one is failing.
```
$ clj -M:cljd flutter
Warming up `.clojuredart/libs-info.edn` (helps us emit better code)

Adding dev dependencies
Resolving dependencies...
+ _fe_analyzer_shared 38.0.0 (41.0.0 available)
+ analyzer 3.4.1 (4.2.0 available)
+ convert 3.0.2
+ crypto 3.0.2
+ file 6.1.2
+ glob 2.1.0
+ package_config 2.1.0
+ pub_semver 2.1.1
  test_api 0.4.11 (0.4.12 available)
+ typed_data 1.3.1
+ watcher 1.0.1
+ yaml 3.1.1
Changed 11 dependencies!

Fetching dependencies
Running "flutter pub get" in hello...                            1,753ms

Dumping type information (it may take a while)
WARNING: could not analyze 'hello' package
Execution error (StackOverflowError) at cljd.compiler/load-libs-info$export$fn (compiler.cljc:134).
null

Full report at:
/tmp/clojure-10647785997140578605.edn
```
Excerpt of the report (I truncated the file, it recurses forever on this `export` function):
```
{:clojure.main/message
 "Execution error (StackOverflowError) at cljd.compiler/load-libs-info$export$fn (compiler.cljc:134).\nnull\n",
 :clojure.main/triage
 {:clojure.error/class java.lang.StackOverflowError,
  :clojure.error/line 134,
  :clojure.error/symbol cljd.compiler/load-libs-info$export$fn,
  :clojure.error/source "compiler.cljc",
  :clojure.error/phase :execution},
 :clojure.main/trace
 {:via
  [{:type java.lang.StackOverflowError,
    :at [clojure.lang.Util hasheq "Util.java" 172]}],
  :trace
  [[clojure.lang.Util hasheq "Util.java" 172]
   [clojure.lang.PersistentHashMap hash "PersistentHashMap.java" 120]
   [clojure.lang.PersistentHashMap
    containsKey
    "PersistentHashMap.java"
    126]
   [clojure.lang.APersistentSet contains "APersistentSet.java" 34]
   [clojure.lang.RT contains "RT.java" 836]
   [clojure.core$contains_QMARK_ invokeStatic "core.clj" 1492]
   [clojure.core$contains_QMARK_ invoke "core.clj" 1484]
   [cljd.compiler$load_libs_info$export__169$fn__172
    invoke
    "compiler.cljc"
    134]
   [clojure.core$keep$fn__8555$fn__8556 invoke "core.clj" 7325]
   [clojure.lang.PersistentVector reduce "PersistentVector.java" 343]
   [clojure.core$transduce invokeStatic "core.clj" 6883]
   [clojure.core$into invokeStatic "core.clj" 6899]
   [clojure.core$into invoke "core.clj" 6887]
   [cljd.compiler$load_libs_info$export__169
    invoke
    "compiler.cljc"
    133]
   [cljd.compiler$load_libs_info$export__169$fn__172
    invoke
    "compiler.cljc"
    137]
   [clojure.core$keep$fn__8555$fn__8556 invoke "core.clj" 7325]
   [clojure.lang.PersistentVector reduce "PersistentVector.java" 343]
   [clojure.core$transduce invokeStatic "core.clj" 6883]
   [clojure.core$into invokeStatic "core.clj" 6899]
   [clojure.core$into invoke "core.clj" 6887]
   [cljd.compiler$load_libs_info$export__169
    invoke
    "compiler.cljc"
    133]
   ...]}}
```
I tried with the head version of clojuredart, same result.
Thanks for the amazing work!

**Comment by perpen** [comment]
Got it working by adding the current lib to `past-exports` in the `shown` (public??) case - but ofc I have no idea what it all means.

**Comment by JasonLee-crypto** [comment]
@perpen I am facing the same issue, how do you got it working, can you give me more information

**Comment by perpen** [comment]
Sure
```
diff --git a/clj/src/cljd/compiler.cljc b/clj/src/cljd/compiler.cljc
index edd6413..0237743 100644
--- a/clj/src/cljd/compiler.cljc
+++ b/clj/src/cljd/compiler.cljc
@@ -134,7 +134,7 @@
                                   (when-not (contains? past-exports lib)
                                     (cond
                                       shown
-                                      (select-keys (export (dart-libs-info lib) lib past-exports) shown)
+                                      (select-keys (export (dart-libs-info lib) lib (conj past-exports lib)) shown)
                                       hidden
                                       (reduce dissoc (export (dart-libs-info lib) lib past-exports) hidden)
                                       :else (export (dart-libs-info lib) lib (conj past-exports lib)))))) exports)]
```

**Comment by JasonLee-crypto** [comment]
Yeah, works like a charm, thanks.

**Comment by dupuchba** [comment]
oh, I thought we got rid of those, can you copy paste your `pubspec.yaml` ?

**Comment by dupuchba** [comment]
fixed cc7f27424c16b86bed8811f9632c1c4e9a59cd81
